lib/hl7/datetime_components.rb

```diff
@@ -37,15 +40,17 @@ def ==(other)
     def validate!
       validate_field!(:year, -9999..9999)
       validate_field!(:month, 1..12)
-      validate_field!(:day, 1..Time.days_in_month(month, year))
+      validate_field!(:day, 1..Time.days_in_month(month, year)) unless day.nil?
```

Comment:
What was the intent of shorting this line out if `day` is `nil`?

Reply:
So let's say we are enforcing the invariant that if `day` is non-nil, then `month` and `year` are also non-nil. Okay, the contrapositive of that is that if `month` or `year` is nil then `day` _must_ be `nil`. In that scenario, when we run this validation, we try to build a range `1..Time.days_in_month(month, year)`, but since one of those arguments to `Time#days_in_month` is `nil`, it blows up.

Anyhow, I'm trying to come up with a way to actually enforce that invariant, and that will affect this code.

Reply:
right, got it. Thanks~